src/Adapter/Order/QueryHandler/GetOrderForViewingHandler.php

```diff
@@ -229,29 +232,30 @@ public function handle(GetOrderForViewing $query): OrderForViewing
      *
      * @return OrderCustomerForViewing|null
      */
-    private function getOrderCustomer(Order $order): ?OrderCustomerForViewing
+    private function getOrderCustomer(Order $order, OrderInvoiceAddressForViewing $invoiceAddress): ?OrderCustomerForViewing
     {
         $currency = new Currency($order->id_currency);
         $customer = new Customer($order->id_customer);
+        $genderName = '';
+        $totalSpentSinceRegistration = null;
 
         if (!Validate::isLoadedObject($customer)) {
-            return null;
-        }
-
-        $gender = new Gender($customer->id_gender);
-        $genderName = '';
+            $this->buildFakeCustomerObject($order, $invoiceAddress, $customer);
```

Comment:
It would be better if the build method returned the customer instead of modifying it